packages/core/src/external/carequality/dq/process-incoming-dq.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  DocumentQueryRequestIncoming,
+  DocumentQueryResponseOutgoing,
+} from "@metriport/ihe-gateway-sdk";
+import {
+  XDSUnknownPatientId,
+  XDSUnknownCommunity,
+  XDSMissingHomeCommunityId,
+  XDSRegistryError,
+  validateDQ,
+} from "./validating-dq";
+import { CODE_SYSTEM_REQUIRED_ERROR as DQ_CODE_SYSTEM_REQUIRED_ERROR } from "../shared";
+
+function constructErrorResponse(
+  payload: DocumentQueryRequestIncoming,
+  codingSystem: string,
+  code: string,
+  error: string
+): DocumentQueryResponseOutgoing {
+  return {
+    id: payload.id,
+    timestamp: payload.timestamp,
+    responseTimestamp: new Date().toISOString(),
+    operationOutcome: {
+      resourceType: "OperationOutcome",
+      id: payload.id,
+      issue: [
+        {
+          severity: "error",
+          code: "processing",
+          details: {
+            coding: [{ system: codingSystem, code: code }],
+            text: error,
+          },
+        },
+      ],
+    },
+  };
+}
+
+export async function processIncomingRequest(
+  payload: DocumentQueryRequestIncoming
+): Promise<DocumentQueryResponseOutgoing> {
+  try {
+    // validate incoming request + look for patient and get all their documents from s3
+    const documentIds = await validateDQ(payload);
+
+    // construct response
+    const response: DocumentQueryResponseOutgoing = {
+      id: payload.id,
+      timestamp: payload.timestamp,
+      responseTimestamp: new Date().toISOString(),
+      extrinsicObjectXmls: documentIds,
```

Comment:
Should we be passing `documentIds` if the key is `extrinsicObjectXmls`. Doesnt seem like those match up